### ISSUE TITLE
Move stylemodifiers and pattern parameters machinery into Mustache engine

### ISSUE THREAD

**Issue by geoffp** [pattern-lab/patternlab-node#615]
Suggested part of https://github.com/pattern-lab/patternlab-node/issues/603.

@bmuenzenmeyer and I have often talked about this, as we believe that Mustache is (and probably will always be) the only engine that uses this functionality. It would be instructive to see how much complexity is subtracted from the core once this is done.

**Comment by bmuenzenmeyer** [comment]
unassigning myself - i only like to assign issues to things i plan to work on imminently

**Comment by JosefBredereck** [comment]
Closing in favor of #1177
